kore/src/main/scala/org/kframework/definition/outer.scala

```diff
@@ -247,13 +248,13 @@ case class Module(val name: String, val imports: Set[Module], localSentences: Se
     Att(union.filter { key => attMap(key._1._1).size == 1 }.toMap)
   }
 
-  lazy val definedSorts: Set[Sort] = (productions map {_.sort}) ++ (sortDeclarations map {_.sort})
+  lazy val definedSorts: Set[Sort] = (productions filter {p => !p.params.contains(p.sort)} map {_.sort}) ++ (sortDeclarations map {_.sort})
```

Comment:
We seem to use this pattern `p.params().contains(p.sort())`, or `p.params().contains(...)`, or `!p.params().contains(...)`. I wonder if it makes sense to factor out a `isSortVariable()` method?